google-cloud-clients/google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryImpl.java

```diff
@@ -507,6 +508,26 @@ public Table getTable(TableId tableId, TableOption... options) {
     return listTables(completeDatasetId, getOptions(), optionMap(options));
   }
 
+  @Override
+  public List<String> listPartitions(TableId tableId) {
+    List<String> partitions = new ArrayList<String>();
+    Table metaTable =
+        getTable(TableId.of(tableId.getDataset(), tableId.getTable() + "$__PARTITIONS_SUMMARY__"));
+    Schema metaSchema = metaTable.getDefinition().getSchema();
+    String partition_id = null;
+    for (Field field : metaSchema.getFields()) {
+      if (field.getName().equals("partition_id")) {
+        partition_id = field.getName();
```

Comment:
Is there any possibility of `Field` to not having `partition_Id` ?

Reply:
If `metaSchema` returns two Field and each of them has different `patition_id`... In that case, we are only considering last received `partition_id`.

Is that correct?

Reply:
> Is there any possibility of `Field` to not having `partition_Id` ?

it has always partition_id

Reply:
> If `metaSchema` returns two Field and each of them has different `patition_id`... In that case, we are only considering last received `partition_id`.
> 
> Is that correct?

metaSchema return total 8 Fields and getting name partition_id which can be used next level.